Classes/Backend/DynamicFlexForm.php

```diff
@@ -162,6 +162,10 @@ public function parseDataStructureByIdentifierPreProcess(array $identifier)
         } else {
             $dataSourceIdentity = $identifier['tableName'] . '.' . $identifier['fieldName'] . ':' . $identifier['uid'];
         }
+        if(!isset(static::$generatedDataSources[$dataSourceIdentity])) {
```

Comment:
I see the issue - identifier is already provided so the hook method above isn't being called when in AJAX mode. It is correct that we need a way to generate the DS in this case.

I would choose to refactor the `getDataStructureIdentifierPreProcess` to separate the logic that generates the data source array, putting that in a protected method which we can call here. That way we don't cause a new identifier to be created and stored.

I can take care of this one later today if you don't want to mess with that refactoring ;)